packages/dataparcels/src/parcel/Parcel.js

```diff
@@ -602,6 +607,29 @@ export default class Parcel {
         }
     };
 
+    _effectUpdate = (effectUpdater: ParcelValueUpdater) => {
+        let {_treeShare} = this;
+        let effectId = `${this.id}-${HashString(effectUpdater.toString())}`;
+
+        // throttle effects with the same effectId
+        // the delay added by throttling is fine because these effects are async anyway
+        if(_treeShare.effectRegistry[effectId]) {
+            return;
+        }
+        _treeShare.effectRegistry[effectId] = true;
+
+        setTimeout(() => {
+            // apply the effect to the current version of the corresponding parcel
+            let parcel = _treeShare.registry[this.id];
+            if(parcel) {
+                // remember to make this action exempt from history
+                // when history is added
+                parcel.update(effectUpdater);
+            }
+            delete _treeShare.effectRegistry[effectId];
+        }, 100);
```

Comment:
What's up with this set timeout? seems like a magic number.

Reply:
It is yeah, but its essentially a debounce / throttle. It's quite like how React has a small delay / buffer to allow very fast subsequent changes to state to be treated as the same state update. I need to bucket all effect calls together to be able to dedupe them, as the thing that calls the effect function can be called multiple times as a change goes up through all the parcels. But I also have no way of knowing when the change is complete and I can stop buffering, because the parcels are all a bit decentralised, there is no god parcel that can say when a round of buffering should stop.